pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -442,3 +443,16 @@ def reconciliate_processes(self, trace: str):
                 os.kill(pid, signal.SIGKILL)
         shutil.rmtree(sync_dir)
         raise DeadlockDetectedException(f"DeadLock detected from rank: {self.global_rank} \n {trace}")
+
+    @contextmanager
+    def block_backward_sync(self):
+        """
+        Blocks ddp sync gradients behaviour on backwards pass.
+        This is useful for skipping sync when accumulating gradients, reducing communication overhead
+        Returns: context manager with sync behaviour off
+        """
+        if isinstance(self.model, DistributedDataParallel):
+            with self.model.no_sync():
+                yield None
+        else:
+            yield None
```

Comment:
n00b question: do you think this ought to be its own mini interface to represent this trait? 

```
class PluginWithBlockBackwardSync(ABC):

    @contextmanager
    @abstractmethod
    def block_backward_sync(self) -> Generator:
```

this way, we only need to check `isinstance(isinstance(self.trainer.training_type_plugin, PluginWithBlockBackwardSync)` in the training batch loop. 

otherwise i'm not sure about the `isinstance` check for custom plugins that require this

(pls ignore the verbose naming)